test/sql/types/nested/map/test_map_from_entries.test

```diff
@@ -0,0 +1,35 @@
+# name: test/sql/types/nested/map/test_map_from_entries.test
+# description: Test basic functionality of map_from_entries function
+# group: [map]
+
+statement ok
+PRAGMA enable_verification
+
+query I
+select MAP_FROM_ENTRIES(ARRAY[(1, 'x'), (2, 'y')]);
+----
+{1=x, 2=y}
+
+# Empty arguments not allowed
+statement error
+SELECT map_from_entries();
+
+# More than one argument
+statement error
+SELECT map_from_entries(ARRAY[(1,2), (3,4)], ARRAY[(5,6), (7,8)]);
+
+# Argument isn't an array
+statement error
+SELECT map_from_entries(5);
+
+# ARRAY doesn't contain structs
+statement error
+SELECT map_from_entries(ARRAY[5,4,3]);
+
+# Struct type contains more than 2 fields (key - value)
+statement error
+select MAP_FROM_ENTRIES(ARRAY[(1, 'x', 'extra'), (2, 'y', 'extra')]);
+
+# Structs inside the ARRAY have different amounts of fields
+statement error
+SELECT MAP_FROM_ENTRIES(ARRAY[(1, 'x'), (2, 'y', 'extra')]);
```

Comment:
In order to test your implementation more thoroughly, also for different types in the list and scenarios, I would add a few more test cases. :)
For example what happens if the list is `NULL`? Or if some of the elements of the list are `NULL`?
```sql
SELECT MAP_FROM_ENTRIES(NULL);
SELECT MAP_FROM_ENTRIES(ARRAY[NULL, (1, 'x'), NULL, (2, 'y')]);
```
Also, try to add some tests where the list parameter is a column, for example you could do this.
```sql
CREATE TABLE t1 AS (SELECT ARRAY[(1, 'x'), (2, 'y')] AS list);
INSERT INTO t1 ... (more rows, also with NULLs)
SELECT MAP_FROM_ENTRIES(list) FROM t1;
```
You can also add test cases covering different types. Does this function also support nested types for keys and/or values, for example this?
```sql
SELECT MAP_FROM_ENTRIES(ARRAY[([1,2], 2), (NULL, NULL), ([3,4], 4), (NULL, NULL), NULL]);
```

Reply:
Thank you for these test cases, they're very useful

In the first case, if a struct is NULL, an InvalidInputException is thrown, maybe instead I should just skip that element ?
The second case with columns works now, I'm setting the validity mask and verified with `validity.AllValid()` that it is actually set to invalid
The last case I don't know what the right behavior is, other than the last element which is NULL, this query succeeds with this as result
```sql
SELECT MAP_FROM_ENTRIES(ARRAY[([1,2], 2), (NULL, NULL), ([3,4], 4), (NULL, NULL)]);
----
{[1, 2]=2, NULL=NULL, [3, 4]=4, NULL=NULL}
```

Reply:
(1) Yes, I don't think that you should throw an exception if an element is `NULL`.
(2) I commented on that test case.
(3) This is a tricky one, you should probably ask someone else on that. You should think about all these cases.
```sql
SELECT MAP_FROM_ENTRIES(ARRAY[([1,2], 2), ([4], NULL), (NULL, 4), (NULL, NULL)]);
```
But I think that this result is actually logical
```
{[1, 2]=2, [4]=NULL, NULL=4, NULL=NULL}
```
I am not so sure about duplicates, but it does not specify anywhere that duplicates are a problem, no?